daemon/src/primary_worker.rs

```diff
@@ -246,6 +246,18 @@ pub async fn spawn_usb_handler(
 
                             devices.insert(serial.clone(), device);
                             change_found = true;
+
+                            if cfg!(target_os = "windows") {
```

Comment:
This doesn't need a runtime OS check, on Linux `.version` will always be 'Some', so the next condition will always return false, which is exactly the same behaviour as this check, just this check is heavier.